frontend/src/metabase/admin/settings/components/SettingsGoogleForm/SettingsGoogleForm.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useRef } from "react";
+import { connect } from "react-redux";
+import { jt, t } from "ttag";
+import MetabaseSettings from "metabase/lib/settings";
+import ActionButton from "metabase/components/ActionButton";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import Breadcrumbs from "metabase/components/Breadcrumbs";
+import {
+  FormField,
+  FormMessage,
+  FormSubmit,
+} from "metabase/containers/FormikForm";
+import { updateGoogleSettings } from "metabase/admin/settings/settings";
+import { settingToFormField } from "metabase/admin/settings/utils";
+import {
+  FormCaption,
+  FormSection,
+  FormHeader,
+  FormRoot,
+} from "./SettingsGoogleForm.styled";
+
+export interface SettingElement {
+  key: string;
+}
+
+export interface SettingsGoogleFormProps {
+  elements?: SettingElement[];
+  settingValues?: Record<string, unknown>;
+  onSubmit: (settingValues: Record<string, unknown>) => void;
+}
+
+const SettingsGoogleForm = ({
+  elements = [],
+  settingValues = {},
+  onSubmit,
+}: SettingsGoogleFormProps) => {
+  const isEnabled = Boolean(settingValues["google-auth-enabled"]);
+  const isEnabledRef = useRef(isEnabled);
+
+  const handleSubmit = (values: Record<string, unknown>) => {
+    return onSubmit({ ...values, "google-auth-enabled": isEnabledRef.current });
+  };
+
+  const handleSaveAndEnableClick = (handleSubmit: () => void) => {
+    isEnabledRef.current = true;
```

Comment:
The thing is that we want to submit the form in 2 ways - one with enabling the auth method and another one not. Our forms framework doesn't support this. The workaround is to track which button was clicked and use this value in the submit handler. Because this happens in one event tick we need to use ref and not state.